File: usacobronze/paint.py
with open("paint.in","r") as fin:
    a,b,c,d = map(int,fin.read().split())
with open("paint.out","w") as fout:
    fout.write(str(max(b,d)-min(a,c) if (a<=c and b>=c or (c<=a and d>=a)) else b-a + d-c))

File: usacobronze/paint.cpp
#include <fstream>
using namespace std;
int main() {
    int a,b,c,d;
    ifstream fin("paint.in");
    fin >> a >> b >> c >> d;
    fin.close();
    ofstream fout("paint.out");
    fout << (a<=c && b>=c || (c<=a && d>=a) ? 
    (b>d? b:d) - (a<c? a:c) // max and min
    :
    b-a+d-c);
}